### ISSUE TITLE
Flow types

### ISSUE THREAD

**Issue by peter-leonov** [nanoutils/nanoutils#163]
Hi guys,

Do you want to do Flow typing yourself or would appreciate a third party help on this topic? 😉

I'm doing Flow in a middle sized project for about a year and typed some crazy functional stuff and may help.

All the best,
Peter.

**Comment by Beraliv** [comment]
@peter-leonov sure help would be appreciated! Especially for Flow

**Comment by Beraliv** [comment]
@peter-leonov what can you suggest now? Are you interested in it?

**Comment by peter-leonov** [comment]
I'll definitely look into after we finish this hard one https://github.com/flow-typed/flow-typed/pull/3012
Flow recently got a breaking change and fixing react-redux took all my type-mana 😂

**Comment by Beraliv** [comment]
@peter-leonov yeah, I've seen your PR, you're doing really impressive work
Keep me posted on your progress

**Comment by Beraliv** [comment]
@peter-leonov congratulations on merged PR and happy holidays!
I am thinking of completing TS types on long Russian weekends
Hope you're ready to make nanoutils better

**Comment by peter-leonov** [comment]
Happy new year and have a short pohmel'e!

During the typing process of `react-redux` I've talked to some people who made me thing that the external type libraries should go. Their idea was that TypeScript won and everything should be written in TypeScript with an exceptions to some real dark magic (kinda like `unsafe` in Rust). And you know what, I tend to agree.

The typing for `react-redux` were a really bad pain point for our team and Flow ecosystem in general, so we've sone it 💪 but this was more of a coincidence than a rule: `react-redux` is written in pure JS and the code base is effectively immutable, so the maintainers, actually, do not care if the library is well typed or not. I hope, with `nanoutils` this is different.

So, in short, I'd say that adding Flow types was a good idea, but there exist better options (trollface).

What do you think?

**Comment by Beraliv** [comment]
@peter-leonov

> Happy new year and have a short pohmel'e!

Ahahah, i have the shortest one now :)

> I've talked to some people who made me thing that the external type libraries should go

I tend to disagree with them
1. I would like to use TypeScript but the library is already written in pure js
2. If we don't use TypeScript, we need to provider correct *.d.ts for better type inference (i.e. VS Code)
3. Also it can be extended with development errors, [Issue#155](https://github.com/nanoutils/nanoutils/pull/155) via [helper function](https://github.com/nanoutils/nanoutils/pull/155/files#diff-0e40c2c70773dd5899b5d210dbb0012fR2) which is defined here: [error/index.js](https://github.com/nanoutils/nanoutils/blob/3be2fc60e001c6e6d8482c38bb8c94c1cd856915/lib/_internal/_error/index.js). It can be implemented better with Flow, [RunTypes](https://github.com/pelotom/runtypes) or anything else, I am open to suggestions

> `react-redux` is written in pure JS and the code base is effectively immutable, so the maintainers, actually, do not care if the library is well typed or not

We have the same situation but:
* Pure JS
* Immutable
* **Care** about types

I am still working on that and hope the option is best

**Comment by Beraliv** [comment]
What do you think after all? @peter-leonov
